core/src/main/resources/error/error-classes.json

```diff
@@ -0,0 +1,22 @@
+{
+  "DIVIDE_BY_ZERO_ERROR" : {
+    "sqlState" : "22012",
+    "messageFormatLines" : [ "divide by zero" ]
+  },
+  "DUPLICATE_KEY_ERROR" : {
+    "sqlState" : "23000",
+    "messageFormatLines" : [ "Found duplicate keys '%s'" ]
+  },
```

Comment:
Do we have `vendorCode`? https://docs.oracle.com/en/java/javase/11/docs/api/java.sql/java/sql/SQLException.html#%3Cinit%3E(java.lang.String,java.lang.String,int)

Reply:
We can add `vendorCode` in the future, but I think we would need to determine how error codes should be defined - whether there should be a class hierarchy, if they should be arbitrary, or something else (eg. hashed from the error class). There are some clients that except the vendor code to match [Hive error code classes](https://issues.apache.org/jira/browse/HIVE-3001), given that they are thrown within a `HiveSQLException`.